scripts/openmc-update-mgxs.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
+"""Update OpenMC's deprecated multi-group cross section XML files to the latest
+HDF5-based format.
+
+Usage information can be obtained by running 'openmc-update-mgxs --help':
+
+usage: openmc-update-mgxs [-h] in out
+
+Update mgxs.xml files to the latest format. This will remove 'outside'
+attributes/elements from lattices and replace them with 'outer' attributes. For
+'cell' elements, any 'surfaces' attributes/elements will be renamed
+'region'. Note that this script will not delete the given files; it will append
+'.original' to the given files and write new ones.
+
+positional arguments:
+  in          Input mgxs xml file
+  out         Output mgxs hdf5 file
+
+optional arguments:
+  -h, --help  show this help message and exit
+
+"""
+
+from __future__ import print_function
+from shutil import move
+import warnings
+import xml.etree.ElementTree as ET
+
+import argparse
+import h5py
+import numpy as np
+
+import openmc.mgxs_library
+
+description = """\
+Update OpenMC's deprecated multi-group cross section XML files to the latest
+HDF5-based format."""
+
+
+def parse_args():
+    """Read the input files from the commandline."""
+    # Create argument parser
+    parser = argparse.ArgumentParser(description=description,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-i', '--input', type=argparse.FileType('r'),
+                        help='input XML file')
+    parser.add_argument('-o', '--output', nargs='?', default='',
+                        help='output file, in HDF5 format')
+    parser.add_argument('-c', '--compression', type=int,
+                        help='HDF5 Compression Level')
+    args = vars(parser.parse_args())
+
+    if args['output'] == '':
+        filename = args['input'].name
+        extension = filename[filename.rfind('.'):]
+        if extension == '.xml':
+            filename = filename[:filename.rfind('.')] + '.h5'
+        args['output'] = filename
+
+    # Parse and return commandline arguments.
+    return args
+
+
+def get_data(element, entry):
+    try:
+        value = element.find(entry).text
+    except:
```

Comment:
Good point.  I revised this to avoid the try/except  completely and made the later try/except more specific with the Exceptions we were worried about.